src/source-af-packet.c

```diff
@@ -1388,14 +1548,168 @@ frame size: TPACKET_ALIGN(snaplen + TPACKET_ALIGN(TPACKET_ALIGN(tp_hdrlen) + siz
     return 1;
 }
 
+static int AFPComputeRingParamsV3(AFPThreadVars *ptv)
+{
+    ptv->req3.tp_block_size = getpagesize();
```

Comment:
It may not be immediately obvious, but the block size in TPACKET_V3 is crucial for true performance tuning. This MUST be configurable, you will want as few polls as possible, and with a page size of e.g. 4096, which is pretty standard, performance gain will actually be quite insignificant. True performance gains with V3 are done by setting the block size (and timeout) as high as you are willing to.
To give you an idea, the block size allows for storing only 2 frames of full MTU size (3x 1500 is already more than the page size). Even with minimum size packets, you will get less than a hundred packets in one block.
Calculate in the interface speed, keeping in mind the preamble and inter packet gap, you will get about 1.4 million packets per second on minimal packet sizes (that's what PF_RING guys are always referring to, because it's the theoretical maximum and will use up more memory per second than bigger frames) and need about 125 MB of memory for a second of 1 GB/s of traffic. Divide this by the timeout, which you chose to be 10 ms, and this means 12.5 kB of data every tick. That's about 4 blocks, which means you will have 3 unnecessary overflows per second at full wire speed. I would rather have much larger blocks and a somewhat higher timeout. At the very least this should be in the configuration, but it will require some documentation.